### PR TITLE
feat: support xrpc custom healthcheck methods

### PR DESCRIPTION
**Pull request by AlinsRan**
By adding callback functions, custom health check methods can be applied to private protocols.

```lua
local checker = healthcheck.new({
  name = "testing",
  shm_name = "test_shm",
  type = "xrpc",
  checks = {
      active = {
          healthy  = {
              interval = 0.5, -- we don't want active checks
              statuses = {200,201},
              successes = 2,
          },
          unhealthy  = {
              interval = 0.5, -- we don't want active checks
              statuses = {500, 501},
              failures = 2,
          },
          xrpc_handler = function(node, conf)
              local http = require('resty.http')
              local httpc = http.new()
              local res, err = httpc:request_uri("http://127.0.0.1:2119/status", {
                  method = "GET",
                  path = "/status",
              })
              if not res then
                  return false
              end
              return true, res.status
          end
      },
  },
})
```

### CODE REVIEW

File: lib/resty/healthcheck.lua [hunk 1]
```diff
@@ -725,6 +730,34 @@ function checker:report_http_status(ip, port, hostname, http_status, check)
...
+      or status == 0 then
+    status_type = "unhealthy"
+    limit = checks.unhealthy.failures
+    ctr = CTR_XRPC
+  else
+    return
+  end
+
+  return incr_counter(self, status_type, ip, port, hostname, limit, ctr)
+
```
Comment (membphis) [inline lib/resty/healthcheck.lua]: code style: remove this line

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@ jobs:
         run: |
           sudo cpanm --notest Test::Nginx > build.log 2>&1 || (cat build.log && exit 1)
           sudo luarocks install luacheck
+          sudo luarocks install api7-lua-resty-http 0.2.2-0
```
Comment (membphis) [inline .github/workflows/ci.yml:30]: we'd better add `xrpc`, it is meaningful

Comment (AlinsRan) [reply .github/workflows/ci.yml:30]: The `resty.http` is only used in testing.
After merging the PR here, the healthcheck library will be released and the xrpc suffix will be added.

Comment (AlinsRan) [reply .github/workflows/ci.yml:30]: image

**Review by membphis — APPROVED** [review]

**Pull request merged**